debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -272,10 +420,8 @@ static void checkSupplementalLogging(OracleConnection connection, String pdbName
                 connection.setSessionToPdb(pdbName);
             }
 
-            final String key = "KEY";
-            String validateGlobalLogging = "SELECT '" + key + "', " + " SUPPLEMENTAL_LOG_DATA_ALL from V$DATABASE";
-            Map<String, String> globalLogging = getMap(connection.connection(false), validateGlobalLogging, UNKNOWN);
-            if ("no".equalsIgnoreCase(globalLogging.get(key))) {
+            Map<String, String> globalLogging = getMap(connection.connection(false), SqlUtils.supplementalLoggingCheckQuery(), UNKNOWN);
```

Comment:
You are absolutely correct. Ideally we have to configure table-level groups for whitelisted columns.
We did it on database level , because of two reasons:
- we wanted to automate it.  Initially I run table level DDLs from the connector code,:
        if (pdbName != null) {
            jdbcConnection.setSessionToPdb(pdbName);
        }

        final String key = "KEY";
        String validateGlobalLogging = "SELECT '" + key + "', " + " SUPPLEMENTAL_LOG_DATA_ALL from V$DATABASE";
        Map<String, String> globalLogging = getMap(connection, validateGlobalLogging, "unknown");
        if ("no".equalsIgnoreCase(globalLogging.get(key))) {
            tableIds.forEach(table -> {
                String tableName = table.schema() + "." + table.table();
                try {
                    String validateTableLevelLogging = String.format("SELECT '%s', LOG_GROUP_TYPE FROM DBA_LOG_GROUPS WHERE LOG_GROUP_TYPE='ALL COLUMN LOGGING' AND OWNER ='%s' AND TABLE_NAME = '%s'", key,
                            table.schema().toUpperCase(), table.table().toUpperCase());
                    Map<String, String> tableLogging = getMap(connection, validateTableLevelLogging, "unknown");
                    if (tableLogging.get(key) == null) {
                        String alterTableStatement = "ALTER TABLE " + tableName + " ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS";
                        LOGGER.info("altering table {} for supplemental logging", table.table());
                        executeCallableStatement(connection, alterTableStatement);
                    }
                } catch (SQLException e) {
                    throw new RuntimeException("Cannot set supplemental logging for table " + tableName, e);
                }
            });
        } else {
            LOGGER.warn("Supplemental logging is set on global level, setting individual table supplemental logging was skipped");
        }

        if (pdbName != null) {
            jdbcConnection.resetSessionToCdb();
        }

 but logminer did not see changes from those tables. I cannot recall why.
- we had no time to experiment with column groups and I set it on the global level for the deadline sake.

Reply:
I think its reasonable to expect that a user has configured the table-level supplemental logging prior to connector start.  I have an open JIRA to look at this, which I plan to investigate this coming week before holiday.